app/services/attribute_asserter.rb

```diff
@@ -0,0 +1,66 @@
+class AttributeAsserter
+  attr_accessor :user, :service_provider, :authn_request
+
+  def initialize(user, service_provider, authn_request)
+    self.user = user
+    self.service_provider = service_provider
+    self.authn_request = authn_request
+  end
+
+  def build
+    attrs = default_attrs
+    add_email(attrs) if bundle.include? 'email'
+    add_mobile(attrs) if bundle.include? 'mobile'
+    add_bundle(attrs) if user.verified?
+    user.asserted_attributes = attrs
+  end
+
+  private
+
+  def default_attrs
+    {
+      uuid: {
+        getter: :uuid,
+        name_format: Saml::XML::Namespaces::Formats::NameId::PERSISTENT,
+        name_id_format: Saml::XML::Namespaces::Formats::NameId::PERSISTENT
+      }
+    }
+  end
+
+  def add_bundle(attrs)
+    bundle.each do |attr|
+      attrs[attr] = { getter: deferred_profile_attribute(attr) }
+    end
+  end
+
+  def deferred_profile_attribute(attr)
+    -> (principal) { principal.active_profile[attr] }
+  end
+
+  def add_email(attrs)
+    attrs[:email] = {
+      getter: :email,
+      name_format: Saml::XML::Namespaces::Formats::NameId::EMAIL_ADDRESS,
+      name_id_format: Saml::XML::Namespaces::Formats::NameId::EMAIL_ADDRESS
+    }
+  end
+
+  def add_mobile(attrs)
+    attrs[:mobile] = { getter: :mobile }
+  end
+
+  def bundle
+    sp_bundle = service_provider.attribute_bundle
+    if sp_bundle.present?
+      sp_bundle
+    elsif authn_request_bundle
+      authn_request_bundle
+    else
+      Profile.default_attribute_bundle
```

Comment:
There is no explicit LOA check. There is a `user.verified?` check. 

Since we are now allowing service providers to completely manage their own attribute bundle definitions, we don't have any kind of above/below-the-line definition of which attributes are "appropriate" for a given LOA level.

So if Agency A wants LOA1 and wants to request `zipcode` the Agency can do that, though we would likely counsel them that (a) the `zipcode` will always be nil unless the user has LOA3 with Agency B, and (b) they probably want LOA3 themselves.